recipe-server/client/control/components/ExtensionForm.js

```diff
@@ -2,18 +2,48 @@ import { Map } from 'immutable';
 import React, { PropTypes as pt } from 'react';
 import { connect } from 'react-redux';
 import { bindActionCreators } from 'redux';
-import { initialize, reduxForm } from 'redux-form';
+import { Field, initialize, reduxForm } from 'redux-form';
+import _ from 'underscore';
 
-import QueryExtension from './data/QueryExtension';
-import { ControlField } from './Fields';
-import { showNotification } from '../actions/NotificationActions';
-import { createExtension, updateExtension } from '../state/extensions/actions';
-import { getExtension } from '../state/extensions/selectors';
+import QueryExtension from 'control/components/data/QueryExtension';
+import { ControlField } from 'control/components/Fields';
+import { showNotification } from 'control/actions/NotificationActions';
+import { createExtension, updateExtension } from 'control/state/extensions/actions';
+import { getExtension } from 'control/state/extensions/selectors';
+
+
+const adaptFileEventToValue = delegate =>
+  e => delegate(e.target.files[0]);
+
+const FileInput = ({
```

Comment:
This might be better suited for the fields modules instead of the extension form module.